src/hotspot/share/code/compressedStream.cpp

```diff
@@ -112,3 +112,106 @@ void CompressedWriteStream::write_long(jlong value) {
   write_signed_int(low(value));
   write_signed_int(high(value));
 }
+
+
+bool CompressedSparseDataReadStream::read_zero() {
+  if (_buffer[_position] & (1 << (7 - _bit_pos))) {
+    return 0; // not a zero data
+  }
+  if (++_bit_pos == 8) {
+    _position++;
+    _bit_pos = 0;
+  }
+  return 1;
+}
+
+uint8_t CompressedSparseDataReadStream::read_byte_impl() {
+  uint8_t b1 = _buffer[_position] << _bit_pos;
+  uint8_t b2 = _buffer[++_position] >> (8 - _bit_pos);
+  return b1 | b2;
+}
+
+jint CompressedSparseDataReadStream::read_int() {
+  if (read_zero()) {
+    return 0;
+  }
+  // integer value encoded as a sequence of 1 to 5 bytes
+  // - the most frequent case (0 < x < 64) is encoded in one byte
+  // - the payload of the first byte is 6 bits, the payload of the following bytes is 7 bits
+  // - the most significant bit in the first byte is occupied by a zero flag
+  // - each byte has a bit indicating whether it is the last byte in the sequence
```

Comment:
in each byte bit #6 indicates whether it is the last byte in the sequence

Reply:
such comment would be misleading